ast/ast.go

```diff
@@ -135,6 +136,15 @@ type RecordSet interface {
 	// Next returns the next row, nil row means there is no more to return.
 	Next() (row types.Row, err error)
 
+	// Read read records into chunk.
+	Read(chk *chunk.Chunk) error
+
+	// NewChunk creates a new chunk with initial capacity.
+	NewChunk() *chunk.Chunk
```

Comment:
Why this need to be a method of the interface?

Reply:
Because we need types to create the chunk, and the types are not easy to construct for the caller.